apps/marginfi-v2-ui/src/components/common/Portfolio/components/move-position/move-position-dialog.tsx

```diff
@@ -0,0 +1,204 @@
+  marginfiAccounts: MarginfiAccountWrapper[];
+  isOpen: boolean;
+  setIsOpen: (isOpen: boolean) => void;
+  bank: ActiveBankInfo;
+  marginfiClient: MarginfiClient | null;
+  fetchMrgnlendState: () => void;
+  extendedBankInfos: ExtendedBankInfo[];
+}) => {
+  const [accountToMoveTo, setAccountToMoveTo] = React.useState<MarginfiAccountWrapper | undefined>(undefined);
+  const [actionTxns, setActionTxns] = React.useState<VersionedTransaction[]>([]);
+  const [isLoading, setIsLoading] = React.useState<boolean>(false);
+
+  const { handleSimulateTxns } = useMoveSimulation({
+    actionTxns,
+    marginfiClient,
+    accountToMoveTo,
+    selectedAccount,
+    activeBank: bank,
+    extendedBankInfos,
+    setActionTxns,
+    setIsLoading,
+  });
+
+  const actionMessages = React.useMemo(() => {
+    const withdrawActionResult = checkLendActionAvailable({
+      amount: bank.position.amount,
+      connected: true,
+      selectedBank: bank,
+      nativeSolBalance: 0,
+      banks: extendedBankInfos,
+      lendMode: ActionType.Withdraw,
+      marginfiAccount: selectedAccount,
+    });
+
+    const depositActionResult = checkLendActionAvailable({
+      amount: bank.position.amount,
+      connected: true,
+      selectedBank: bank,
+      nativeSolBalance: 0,
+      banks: extendedBankInfos,
+      lendMode: ActionType.Deposit,
+      marginfiAccount: accountToMoveTo!,
+    });
+
+    return [...withdrawActionResult, ...depositActionResult];
+  }, [bank, selectedAccount, extendedBankInfos, accountToMoveTo]);
+
+  const isButtonDisabled = React.useMemo(() => {
+    if (!accountToMoveTo) return true;
+    if (actionMessages && actionMessages.filter((value) => value.isEnabled === false).length > 0) return true;
+    if (isLoading) return true;
+    return false;
+  }, [accountToMoveTo, actionMessages, isLoading]);
+
+  const handleMovePosition = React.useCallback(async () => {
+    if (!marginfiClient || !accountToMoveTo || !actionTxns) {
+      return;
+    }
+
+    const multiStepToast = new MultiStepToastHandle("Moving position", [
+      {
+        label: `Moving to account ${`${accountToMoveTo?.address.toBase58().slice(0, 8)}
+          ...${accountToMoveTo?.address.toBase58().slice(-8)}`}`,
+      },
+    ]);
+    multiStepToast.start();
+    setIsLoading(true);
+    try {
+      await marginfiClient.processTransactions(actionTxns);
+      await fetchMrgnlendState();
+      multiStepToast.setSuccessAndNext();
+      setIsOpen(false);
+    } catch (error) {
+      console.error("Error moving position between accounts", error);
+      multiStepToast.setFailed("Error moving position between accounts"); // TODO: update
+    } finally {
+      setIsLoading(false);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [marginfiClient, accountToMoveTo, selectedAccount, bank]);
+
+  React.useEffect(() => {
+    if (!accountToMoveTo) return;
+    handleSimulateTxns();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
same here, shouldn't handleSimulateTxns be triggered if the function updated? or does `react-hooks/exhaustive-deps` have a good reason of being here